brewtils/plugin.py

```diff
@@ -665,6 +664,47 @@ def _validate_system(self):
                 "@system(bg_version=%s)" % (self._system.version, client_version)
             )
 
+    def _find_logger_basefilename(self, logger):
+        """Finds the logger base filename(s) currently there is only one
+        """
+        log_file = None
+        parent = logger.parent
+        if parent.__class__.__name__ == "RootLogger":
+            # this is where the file name lives
+            for h in parent.handlers:
+                if hasattr(h, "baseFilename"):
+                    log_file = h.baseFilename
+                    break
+        else:
+            log_file = self._find_logger_basefilename(parent)
+
+        return log_file
+
+    def _read_log(self, start_line=0, end_line=20, read_all=False, read_tail=False):
+
+        log_file = self._find_logger_basefilename(logging.getLogger(__name__))
+        if log_file:
+            try:
+                with open(log_file, "r") as text_file:
+                    raw_logs = text_file.readlines()
+                    if read_all:
+                        return raw_logs
+                    elif read_tail:
+                        return raw_logs[(len(raw_logs) - start_line) : :]
+                    else:
+                        return raw_logs[start_line:end_line]
```

Comment:
Ehh, nevermind about this. The default config is to rotate log files after 10MB. If they have a log that's so large it can't fit in memory they should probably tweak their config.